### ISSUE TITLE
How Do I Remove Edge

### ISSUE THREAD

**Issue by JesusIslam** [solher/arangolite#12]
Is this even possible using AQL only? 

[From here](http://stackoverflow.com/questions/35074545/safe-removal-of-vertexes-in-arangodb-using-ids) it says that currently there is no way to do this via AQL. But arangolite doesn't seems to have a way to remove edge using graph management interface?

**Comment by solher** [comment]
To be honest, I also don't really understand why the ArangoDB team never implemented some kind of delete cascade.

The [remove vertex operation](https://docs.arangodb.com/GeneralGraphs/Management.html#remove-a-vertex) from the graph management interface is Javascript so I would not consider it as a viable option in a Go program (but I did it with the transactions so why not...).
The Javascript API is still way ahead of AQL (for example the transactions that can only be done using Javascript right now) but they are working hard on implementing everything natively for AQL and ArangoDB 3.0 should be a great step towards that.

**Comment by JesusIslam** [comment]
Yea, it feels really incomplete right now when AQL doesn't have the same capability as the JS interface.

I think I should get the edge first somehow to get the key, then do a `Send` to `DELETE /_api/edge/{col}/{key}`.

Thought?

**Comment by solher** [comment]
Actually, the best solution I found for now is using the AQL traversals like that:

```go
	t := arangolite.NewTransaction(
		nil,
		[]string{"users","dashboards","dashboardRights","devices","deviceRights"},
     ).AddQuery("users", `
		FOR u IN users
		REMOVE u IN users
		RETURN u
	`).AddQuery("", `
		FOR u IN {{.users}}
			FOR v,e IN OUTBOUND u dashboardRights, deviceRights
			REMOVE v IN dashboards OPTIONS { ignoreErrors: true }
			REMOVE e IN dashboardRights OPTIONS { ignoreErrors: true }
			REMOVE v IN devices OPTIONS { ignoreErrors: true }
			REMOVE e IN deviceRights OPTIONS { ignoreErrors: true }
	`).Return("users")
```

The thing which is not cool is that we have to statically indicate the collection in which we want to remove a vertex/edge. So here we have basically to try every possibility. Still not very beautiful haha.

**Comment by JesusIslam** [comment]
Seems complicated for a task so simple.

I think I need to learn AQL more to totally understand what's going on in that query.

**Comment by solher** [comment]
Yeah I know and I totally agree with you. There should be some simple way to cascade when there's a deletion...
